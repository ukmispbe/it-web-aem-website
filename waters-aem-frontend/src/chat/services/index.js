import 'whatwg-fetch';

class ChatService {
    constructor(
        countryCode = 'US',
        statusApi = 'https://stgservices.waters.com/api/waters/v1/chat/enabled/{countryCode}'
    ) {
        this.countryCode = countryCode;
        this.statusApi = statusApi;
    }

    checkFetch(response) {
        if (!response.ok){
            throw response;
        }
        return response;
    }

    getChatStatus() {
        const url = this.statusApi.replace('{countryCode}', this.countryCode);
        return this.getData(url);
    }

    getData(url){
        return new Promise((resolve, reject) => {
            window
                .fetch(url)
                .then(this.checkFetch)
                .then(response => {
                    resolve(response.json())
                })
                .catch(err => {
                    reject(err);
                });
        });
    }
}

export default ChatService;