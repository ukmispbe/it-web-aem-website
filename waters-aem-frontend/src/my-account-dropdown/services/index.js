import 'whatwg-fetch';
import SessionStore from '../../stores/sessionStore';
import LocalStore from '../../stores/localStore';
import {setUserDataForDataDog} from '../../utils/userFunctions'

const getData = async (url) => {
    const response = await fetch(url, {
        method: 'GET',
        credentials: 'include',
        headers: {
            'Content-Type': 'application/json'
        }
    });
    return response;
};

export async function signOutRequest(url, signOutUrl, homepageLink) {
    const sessionStore = new SessionStore();
    sessionStore.removeUserDetails();
    sessionStore.removeSoldToDetails();
    const localStore = new LocalStore();
    localStore.removeGUID();
    localStore.removeCartId();

    const response = await getData(url);
    //TODO: Remove after MyAccount-R-6.1.0 release

    if(signOutUrl.indexOf('/nextgen') === -1) {
        homepageLink = signOutUrl;
    }
    setUserDataForDataDog();
    window.location.href = homepageLink;
}