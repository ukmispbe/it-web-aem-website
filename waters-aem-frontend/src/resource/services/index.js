/**
 * Builds Resource data
 * @param {Array} data 
 * @returns Array
 */
const formData = data => {
    const { num_found } = data;
    if (num_found > 0) {
        const { documents } = data;
        return documents.reduce((acc, item) => {
            acc.push({
                title: item.title,
                url: item.url,
                yearpublished_facet: item.yearpublished_facet ? item.yearpublished_facet.replace('[', '').replace(']', '') : ''
            });
            return acc;
        }, []);
    } else {
        return [];
    }
}

/**
 * Fetches data from given url
 * @param {String} url 
 * @returns {Object} 
 */
const getResourcesData = async url => {
    const callService = window.fetch(url).then(res => res.json());
    const data = await callService;
    const response = { numFound: data.num_found };
    response.documents = formData(data);
    return response;
};

export default getResourcesData;