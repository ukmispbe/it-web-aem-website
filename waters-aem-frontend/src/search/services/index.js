import 'whatwg-fetch';
import SessionStore from '../../stores/sessionStore';
import { getCategoryReferenceType } from '../../utils/userFunctions';

const queryString = require('query-string');

const parameterValues = {
    undefined: 'undefined',
    sort: {
        mostRecent: 'most-recent',
        mostRelevant: 'most-relevant',
    },
};

const parameterDefaults = {
    page: 1,
    rows: 25,
    keyword: '*:*',
    category: '',
    content_type: '',
    sort: parameterValues.sort.mostRecent,
    selectedFacets: {},
    contentTypeSelected: {},
};

class SearchService {
    constructor(
        isocode,
        path = 'https://stgservices.waters.com/api/waters/search',
        page = parameterDefaults.page,
        rows = parameterDefaults.rows,
        sort = parameterDefaults.sort,
        multiselect = true,
        throwError
    ) {
        this.path = path;
        this.options = {
            isocode,
            page,
            rows,
            sort,
            multiselect,
        };
        this.throwError = throwError;
        this.sessionStore = new SessionStore();
    }

    getCategories = ({
        keyword = parameterDefaults.keyword,
        page = parameterDefaults.page,
        sort = parameterDefaults.sort,
    }) => {
        const paramString = this.getQueryParamString({ keyword, page, sort });
        const searchString = `${this.path}?${paramString}`;

        return window
            .fetch(searchString)
            .then(response => {
                if (response.ok) {
                    return response.json();
                } else {
                    this.throwError(response);
                }
            })
            .catch(err => console.log(err));
    };

    getResultsByCategory = ({
        keyword = parameterDefaults.keyword,
        facets = {},
        page = parameterDefaults.page,
        sort = parameterDefaults.sort,
        category = parameterDefaults.category,
    } = {}) => {
        const paramString = this.getQueryParamString({ keyword, page, sort });
        const searchString = `${
            this.path
        }/category_facet$${category.toLowerCase()}:${encodeURIComponent(
            encodeURIComponent(category)
        )}?${paramString}`;

        return window
            .fetch(searchString)
            .then(response => {
                if (response.ok) {
                    return response.json();
                } else {
                    this.throwError(response);
                }
            })
            .catch(err => console.log(err));
    };

    getContentType = (
        contentTypeKey,
        contentTypeValue,
        {
            keyword = parameterDefaults.keyword,
            facets = {},
            page = parameterDefaults.page,
            sort = parameterDefaults.sort,
            category = parameterDefaults.category,
        } = {}
    ) => {
        const paramString = this.getQueryParamString({ keyword, page, sort });
        const searchString = `${
            this.path
        }/category_facet$${category.toLowerCase()}:${encodeURIComponent(
            encodeURIComponent(category)
        )}&contenttype_facet$${contentTypeKey}:${encodeURIComponent(
            encodeURIComponent(contentTypeValue)
        )}?${paramString}`;

        return window.fetch(searchString).then(response => {
            if (response.ok) {
                return response.json();
            } else {
                this.throwError(response);
                return response;
            }
        });
    };

    getSubFacet = (
        contentTypeName,
        contentTypeValue,
        {
            keyword = parameterDefaults.keyword,
            facets = {},
            page = parameterDefaults.page,
            sort = parameterDefaults.sort,
            category = parameterDefaults.category,
        } = {}
    ) => {
        const paramString = this.getQueryParamString({ keyword, page, sort });
        const facetString = this.getQueryFacetString(facets);
        const searchString = `${
            this.path
        }/category_facet$${category.toLowerCase()}:${encodeURIComponent(
            encodeURIComponent(category)
        )}&contenttype_facet$${contentTypeName.replace(
            '_facet',
            ''
        )}:${encodeURIComponent(
            encodeURIComponent(contentTypeValue)
        )}${facetString}?${paramString}`;

        return window.fetch(searchString).then(response => {
            if (response.ok) {
                return response.json();
            } else {
                this.throwError(response);
                return response;
            }
        });
    };

    getSuggestedKeywords = async (rows, term) => {
        const searchString = `${this.path}/v1/autocomplete?term=${term}&rows=${rows}&isocode=${this.options.isocode}${getCategoryReferenceType()}`;

        const callService = window.fetch(searchString).then(response => {
            if (response.ok) {
                return response.json();
            } else {
                this.throwError(response);
                return response.json();
            }
        });

        const response = await callService;

        return response.suggestions;
    };

    getParamsFromString() {
        const str = window.location.search;
        const obj = queryString.parse(str);
        obj.selectedFacets = {};

        if (Array.isArray(obj.facet)) {
            for (let i = 0; i < obj.facet.length; i++) {
                const facetSplit = obj.facet[i].split(':');
                const decodedFacetValue = decodeURIComponent(facetSplit[1]);

                if (!obj.selectedFacets[facetSplit[0]]) {
                    obj.selectedFacets[facetSplit[0]] = [];
                }

                obj.selectedFacets[facetSplit[0]].push(decodedFacetValue);
            }
        } else {
            if (obj.facet) {
                const facetSplit = obj.facet.split(':');
                const decodedFacetValue = decodeURIComponent(facetSplit[1]);

                if (!obj.selectedFacets[facetSplit[0]]) {
                    obj.selectedFacets[facetSplit[0]] = [];
                }

                obj.selectedFacets[facetSplit[0]].push(decodedFacetValue);
            }
        }

        return obj;
    }

    getQueryParamString(
        {
            keyword = parameterDefaults.keyword,
            page = parameterDefaults.page,
            sort = parameterDefaults.sort,
            category = parameterDefaults.category,
            content_type = parameterDefaults.content_type,
        } = {},
        facets
    ) {
        const fullParams = Object.assign({}, this.options, {
            keyword,
            page,
            sort,
            category,
            content_type,
        });

        if (!fullParams.category) delete fullParams.category;

        if (!fullParams.content_type) delete fullParams.content_type;

        let paramString = queryString.stringify(fullParams);

        if (facets) {
            for (let i = 0; i <= Object.keys(facets).length; i++) {
                const key = Object.keys(facets)[i];
                const facet = facets[key];

                if (facet) {
                    for (let n = 0; n < facet.length; n++) {
                        const f = encodeURIComponent(facet[n]);
                        paramString =
                            paramString + `&facet=${key}:${encodeURI(f)}`;
                    }
                }
            }
        }

        return paramString;
    }

    getQueryFacetString(facets) {
        let facetString = '';

        for (let i = 0; i <= Object.keys(facets).length; i++) {
            const facetName = Object.keys(facets)[i];
            const category = facetName
                ? `${facetName}$${facetName.replace('_facet', '')}`
                : null;
            const facet = facets[facetName];

            if (facet && category) {
                if (i === 0) {
                    facetString = facetString + `&${category}:`;
                } else {
                    facetString = facetString + `&${category}:`;
                }

                for (let f = 0; f <= facet.length; f++) {
                    const filter = facet[f];

                    if (filter) {
                        facetString = filter
                            ? facetString +
                            `${
                                f > 0 ? encodeURIComponent('||') : ''
                            }${encodeURIComponent(encodeURIComponent(filter))}`
                            : facetString;
                    }
                }
            } else if (category) {
                facetString = facetString + `&${category}:`;
            }
        }

        return facetString;
    }

    createQueryObject(params) {
        const obj = {};

        obj['keyword'] = params.keyword;
        obj['page'] = params.page || parameterDefaults.page;
        obj['facets'] = {};
        obj['sort'] = params.sort;

        if (params.category) obj['category'] = params.category;

        if (params.content_type) obj['content_type'] = params.content_type;

        if (params.facet) {
            const facets = params.facet;

            if (Array.isArray(facets)) {
                for (let n = 0; n <= facets.length; n++) {
                    const facet = facets[n];
                    if (facet) {
                        const splitName = facet.split(':');
                        if (Array.isArray(obj['facets'][splitName[0]])) {
                            obj['facets'][splitName[0]].push(decodeURIComponent(splitName[1]));
                        } else {
                            obj['facets'][splitName[0]] = [decodeURIComponent(splitName[1])];
                        }
                    }
                }
            } else if (facets) {
                const splitName = facets.split(':');
                obj['facets'][splitName[0]] = [decodeURIComponent(splitName[1])];
            }
        }

        return obj;
    }

    getUrlParameter = (sParam, sPageURL) => {
        const sURLVariables = sPageURL.split('&');

        for (let i = 0; i < sURLVariables.length; i++) {
            const sParameterName = sURLVariables[i].split('=');

            if (sParameterName[0] === sParam) {
                return sParameterName[1] === undefined
                    ? true
                    : decodeURIComponent(sParameterName[1]);
            }
        }
    };

    mapFacetGroupsToArray = facetParam => {
        if (Array.isArray(facetParam)) {
            const facetArray = facetParam.map(item => {
                const splitArray = item.split(':');
                return splitArray.length === 2 ? splitArray[0] : '';
            });

            return facetArray.filter(item => item !== '');
        } else if (facetParam) {
            const splitArray = facetParam.split(':');
            return splitArray.length === 2 ? [splitArray[0]] : [];
        }

        return [];
    }

    buildParameters = searchValue => {
        const keyword = searchValue ? searchValue : parameterDefaults.keyword;
        const sort =
            keyword === parameterDefaults.keyword
                ? parameterDefaults.sort
                : parameterValues.sort.mostRelevant;

        return { keyword, sort };
    };

    stringifyParameters = parameters =>
        Object.keys(parameters).length !== 0
            ? Object.keys(parameters).reduce(
                (accumulator, currentValue) =>
                    `${accumulator}=${parameters[accumulator]}&${currentValue}=${parameters[currentValue]}`
            )
            : '';

    setUrlParameter = (searchTerm, searchPath) => {
        const parameters = this.buildParameters(searchTerm);
        const querystring = this.stringifyParameters(parameters);

        window.location.href = `${searchPath}?${querystring}`;
    };

    isDefaultKeyword = value => value === parameterDefaults.keyword;

    setStorageForPagePosition = () => {
        const scrolled =
            (window.pageYOffset || window.document.scrollTop) -
            (window.document.clientTop || 0);

        this.sessionStore.setPreviousPagePosition(scrolled);
        this.sessionStore.setFromSearchURL(window.location.href);
    }

    setStorageForTabHistory = tabHistory => {
        this.sessionStore.setSearchTabHistory(tabHistory);
    }

    setStorageForPagination = () => {
        const scrolled =
            (window.pageYOffset || window.document.scrollTop) -
            (window.document.clientTop || 0);
        
        this.sessionStore.setPreviousPaginationClick(scrolled);
    }

    getSessionStore = () => {
        const previousPagePosition = 
            this.sessionStore.getPreviousPagePositionEnabled()
            ? this.sessionStore.getPreviousPagePosition()
            : null;

        return {
            previousPagePosition,
            fromSearchURL: this.sessionStore.getFromSearchURL(),
            searchTabHistory: this.sessionStore.getSearchTabHistory(),
            previousPaginationClick: this.sessionStore.getPreviousPaginationClick()
        }
    }

    clearSessionStore = () => {
        this.sessionStore.removePreviousPagePosition();
        this.sessionStore.removePreviousPagePositionEnabled();
        this.sessionStore.removeFromSearchURL();
        this.sessionStore.removeSearchTabHistory();
        this.sessionStore.removePreviousPaginationClick();
    }

    scrollToPosition = position => {
        window.scrollTo(0, position);
        this.sessionStore.removePreviousPagePosition();
        this.sessionStore.removePreviousPagePositionEnabled();
    }

    scrollToTop = () => {
        window.scrollTo(0, 0);
        this.sessionStore.removePreviousPagePositionEnabled();
    }
}

const searchMapper = {
    mapFacetGroups: (contentType, filterMap, facets) => {
        const facetName = `${contentType}_facet`;

        const facet = Array.isArray(filterMap.orderedFacets)
            ? filterMap.orderedFacets.find(item => item.facetName === facetName)
            : null;

        if (!facet) { return; }

        const orderedFacets = facet.orderedFacets.filter(item => facets[item.facetName]);

        const mapping = orderedFacets.map(facet => {
            return {
                name: facet.facetName,
                category: facet.facetValue,
                translation: facet.facetTranslation,
                facets: facets[facet.facetName],
                isExpanded: false,
            }
        });

        return mapping;
    }
};

export { SearchService, parameterValues, parameterDefaults, searchMapper };