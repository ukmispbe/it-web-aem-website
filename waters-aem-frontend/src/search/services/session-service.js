const sessionKeys = {
    searchTerm: 'searchTerm'
}

export default class SessionService {
    setSearchTerm = value => sessionStorage.setItem(sessionKeys.searchTerm, value);
    getSearchTerm = () => sessionStorage.getItem(sessionKeys.searchTerm);
    removeSearchTerm = () => sessionStorage.removeItem(sessionKeys.searchTerm);
}