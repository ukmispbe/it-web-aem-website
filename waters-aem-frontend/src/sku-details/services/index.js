import 'whatwg-fetch';
import LocalStore from '../../stores/localStore';
import loginStatus from '../../scripts/loginStatus';
import { fetchData } from '../../utils/serviceFunctions';
import { getCountryCode, getLanguage, getUserId } from '../../utils/userFunctions';
import { isEprocurementUser, getEprocUserCountryCode, getEprocUserLanguage } from '../../utils/userFunctions';

const availabilityUrlRequest = (url, countryCode, partNo) => {
    url = url
            .replace('{partnumber}', partNo)
            .replace('{countryCode}', isEprocurementUser() ? getEprocUserCountryCode().toUpperCase() : countryCode);

    return url;
}

const priceUrlRequest = (endpoint, sku, soldToId, salesOrg) => {
    let url;
    return url = `${endpoint}?productNumber=${sku}&customerNumber=${soldToId}&salesOrg=${salesOrg}`;
}

const legacyAddToCartUrlRequest = (url, partNo, quantity) => {
    url = url
        .replace('{partnumber}', partNo)
        .replace('{quantity}', quantity);

    return url;
}

const addToCartUrlRequest = (url, partNo, quantity, cartId) => {
    let userId = getUserId();
    userId = userId !== '' ? userId : 'anonymous';

    url = url
        .replace('{localeCountry}', isEprocurementUser() ? getEprocUserCountryCode().toLowerCase() : getCountryCode())
        .replace('{localeLanguage}', isEprocurementUser() ? getEprocUserLanguage().toLowerCase() : getLanguage())
        .replace('{userType}', userId)
        .replace('{guid}', cartId ? cartId : 'null')
        .concat('', '?successWithCart=true');
    url = cartId ? url : url.concat('', `&createCart=${!isEprocurementUser()}`);

    return url;
}

export async function addToCart(isCommerceApiMigrated, url, partNo, quantity, throwError) {
    if(isCommerceApiMigrated === 'true' || isCommerceApiMigrated === true) {
        // Check if partNo is a single product or an array of products
        let products = '';
        if (Array.isArray(partNo)) {
			products = {
				products: partNo,
			};
			} else {
			products = {
				products: [
				{
					code: partNo,
					quantity: quantity,
				},
				],
			};
        }
        const options = {
            method: 'POST',
            credentials: 'include',
            mode: 'cors',
            headers: {
                'Content-Type': 'application/json'
            },
            body: JSON.stringify(products)
        }
        const localStore = new LocalStore();
        const cartId = loginStatus.state() ? localStore.getCartId() : localStore.getGUID();

        const urlRequest = addToCartUrlRequest(url, partNo, quantity, cartId);
        const response = await fetchData(urlRequest, options, throwError);
        if(response.status === 200) {
            const json = await response.json();
            if(!cartId && json) {
                loginStatus.state() && json.cart.code && localStore.setCartId(json.cart.code);
                !loginStatus.state() && json.cart.guid && localStore.setGUID(json.cart.guid);
            }
            return json;

        } else if(response.status === 400) {
            const json = await response.json();
            // if cartId or guid is no longer valid
            if(json && json.errors && json.errors.length && json.errors[0].type === 'CartError') {
                loginStatus.state() && cartId && localStore.removeCartId();
                !loginStatus.state() && cartId && localStore.removeGUID();
                addToCart(isCommerceApiMigrated, url, partNo, quantity, throwError);
            }
            else{
                throwError({status: 500, ok: false});
                return response.status;
            }
        } else {
            throwError({status: 500, ok: false});
            return response.status;
        }

    } else {

        const options = {
            method: 'POST',
            credentials: 'include',
            body: JSON.stringify({
                partNumbers: partNo,
                quantity: quantity,
            })
        }
        const urlRequest = legacyAddToCartUrlRequest(url, partNo, quantity);
        const response = await fetchData(urlRequest, options, throwError);
        const json = await response.json();
        return json;
    }
}

export async function getAvailability(url, countryCode, partNo) {
    const options = {
        method: 'GET',
        credentials: 'include'
    }

    const urlRequest = availabilityUrlRequest(url, countryCode, partNo);
    const response = await fetchData(urlRequest, options);
    const json = await response.json();
    return json;
}

export async function getPricing(url, sku, soldToId, salesOrg) {
    if (Array.isArray(sku)) {
        sku = sku.map(skuItem => skuItem.code).join(',');
    }

    const options = {
        method: 'GET',
        credentials: 'include',
        mode: 'cors'
    }

    const urlRequest = priceUrlRequest(url, sku, soldToId, salesOrg);
    const response = await fetchData(urlRequest, options);
	const json = await response.json();

	if(response.status === 200) {
        json.status = 200;	
	} else {	
		json.status = response.status;
	}
	return json;
}

export const matchListItems = (skuListData, pricesAPIResults) => {
let skuListItem = {
	code: skuListData
}
	for (let i = 0; i < pricesAPIResults.length; i++) {
		if(skuListItem.code === pricesAPIResults[i].productNumber) {
			skuListItem.custPrice = pricesAPIResults[i].netPrice.formattedValue;
			skuListItem.custValue = pricesAPIResults[i].netPrice.value;
			skuListItem.listPrice = pricesAPIResults[i].basePrice.formattedValue;
			skuListItem.listValue = pricesAPIResults[i].basePrice.value;
			skuListItem.currencyCode = pricesAPIResults[i].netPrice.currencyCode;
		} 
	}

    return skuListItem;
}