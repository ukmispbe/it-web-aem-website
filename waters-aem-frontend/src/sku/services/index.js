import 'whatwg-fetch';

class SkuService {
    constructor(
        isocode,
        sku = {
            path: 'https://dev-www.waters.com:8443/api/waters/',
            endpoints: {
                availability: 'availability/',
                customerPrice: 'customerprice/',
            },
        },
        addToCart = {
            path:
                'https://www.waters.com/waters/ajax.htm?handler=shoppingHandler&action=processExternalCart',
        },
        throwError
    ) {
        this.isocode = isocode;
        this.skuOptions = sku;
        this.addToCartOptions = addToCart;
        this.throwError = throwError;
    }

    getAvailability(partNo) {
        return new Promise((resolve, reject) => {
            window
                .fetch(this.createAvailabilityRequest(partNo))
                .then(response => {
                    if (response.ok) {
                        resolve(response.json());
                    } else {
                        this.throwError(response);
                        reject(response);
                    }
                });
        });
    }

    createAvailabilityRequest(partNo) {
        const path = `${this.skuOptions.path}/product/v1/`;
        const endpoint = this.skuOptions.endpoints.availability;
        const request = `${path}${endpoint}/${partNo}/${this.isocode}`;

        return request;
    }

    getPrice(partNo) {
        return new Promise((resolve, reject) => {
            window.fetch(this.createPriceRequest(partNo)).then(response => {
                if (response.ok) {
                    resolve(response.json());
                } else {
                    this.throwError(response);
                    reject(response);
                }
            });
        });
    }

    createPriceRequest(partNo) {
        const path = `${this.skuOptions.path}/product/v1/`;
        const endpoint = this.skuOptions.endpoints.customerPrice;
        const request = `${path}${endpoint}/${partNo}/${this.isocode}`;

        return request;
    }

    addToCart(partNo, quantity) {
        const path = this.addToCartOptions.path;
        return new Promise((resolve, reject) => {
            window
                .fetch(path, {
                    method: 'post',
                    body: JSON.stringify({
                        partNumbers: partNo,
                        quantity: quantity,
                    }),
                })
                .then(response => {
                    if (response.ok) {
                        resolve(response.json());
                    } else {
                        this.throwError(response);
                        reject(response);
                    }
                });
        });
    }
}

export default SkuService;
