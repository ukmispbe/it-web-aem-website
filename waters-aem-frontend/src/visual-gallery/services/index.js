import "whatwg-fetch";

const getBrightCoveApi = (brightCoveApi, accountId, videoId) => {
  return  brightCoveApi.replace(/{accountId}/gi, accountId).replace(/{videoId}/gi, videoId);    
}

export const getData = (url, policyKey) => {
    return new Promise((resolve, reject) => {
        window
            .fetch(url,  
                {
                    method: "GET",
                    headers: {
                      Accept: `application/json;pk=${policyKey}`,
                    },
                }
            )
            .then(response => {
                resolve(response.json())
            })
            .catch(err => {
                reject(err);
            });
    });
}

export const getBrightCoveVideoData = async (videoIds, accountId, brightCoveApi, policyKey, cb) => {
    const promises = [];
    videoIds.map( videoId => promises.push(getData(getBrightCoveApi(brightCoveApi, accountId, videoId),policyKey)))
    Promise.all(promises).then(response => cb(response));
}