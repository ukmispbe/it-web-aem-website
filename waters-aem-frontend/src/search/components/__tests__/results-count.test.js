import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { parameterDefaults } from '../../services';
import ResultsCount from '../results-count';

describe('Feature: ResultsCount React Component', () => {
    const props = {
        rows: parameterDefaults.rows,
        count: 0,
        query: parameterDefaults.keyword,
        current: parameterDefaults.page,
        noQuery: true,
        spell_related_suggestions: [],
        spell_suggestion: '',
        onRelatedSuggestionClick: jest.fn(() => {}),
        text: {
            resultsText: 'Showing {count} results',
            resultsForText: 'Showing {count} results for ',
            noResultsText: 'Showing 0 results',
            relatedSearchesText: 'Related Searches',
            inCategoryText: 'in ',
            searchIcon: 'search.svg'
        },
        categoryOptions: [
            {
                translation: "Shop",
                name: "Shop",
                count: 633
            }
        ],
        categoryValue: 0
    };

    describe('Scenario: Rendering', () => {
        describe('When the number of results is higher than the allowed page count', () => {
            it('Then the snapshot should match', () => {
                props.count = props.rows + 1;

                const json = renderer.create(<ResultsCount {...props} />);
                expect(json).toMatchSnapshot();
            });
        });

        describe('When the number of results is less than the allowed page count', () => {
            it('Then the snapshot should match', () => {
                props.count = props.rows - 1;

                const json = renderer.create(<ResultsCount {...props} />);
                expect(json).toMatchSnapshot();
            });
        });

        describe('When keyword is not provided', () => {
            it('Then the snapshot should match', () => {
                props.noQuery = false;
                props.query = parameterDefaults.keyword;

                const json = renderer.create(<ResultsCount {...props} />);
                expect(json).toMatchSnapshot();
            });
        });

        describe('When keyword is provided without spelling suggestion', () => {
            it('Then the snapshot should match', () => {
                props.query = 'milk';

                const json = renderer.create(<ResultsCount {...props} />);
                expect(json).toMatchSnapshot();
            });
        });

        describe('When keyword is provided with spelling suggestion', () => {
            it('Then the snapshot should match', () => {
                props.query = 'milk';
                props.spell_suggestion = 'got milk?';

                const json = renderer.create(<ResultsCount {...props} />);
                expect(json).toMatchSnapshot();
            });
        });

        describe('When one related keyword is provided', () => {
            it('Then the snapshot should match', () => {
                props.query = 'milk';
                props.spell_related_suggestions = ['... it does a body good!'];

                const json = renderer.create(<ResultsCount {...props} />);
                expect(json).toMatchSnapshot();
            });
        });

        describe('When more than one related keyword is provided', () => {
            it('Then the snapshot should match', () => {
                props.query = 'milk';
                props.spell_related_suggestions = ['... it does a body good!', 'milkyway'];

                const json = renderer.create(<ResultsCount {...props} />);
                expect(json).toMatchSnapshot();
            });
        });
    });

    describe('Scenario: User Interaction', () => {
        describe('When related suggestion is clicked', () => {
            it('Then the props click event handler should be called', () => {
                props.query = 'milk';
                props.spell_related_suggestions = ['... it does a body good!'];
                
                const spyOnHanlder = spyOn(props, 'onRelatedSuggestionClick');

                const wrapper = shallow(<ResultsCount {...props} />);

                wrapper.find('.cmp-search__related-suggestions .item').simulate('click');

                expect(spyOnHanlder).toHaveBeenCalled();

                spyOnHanlder.mockRestore();
            });
        });
    });
});