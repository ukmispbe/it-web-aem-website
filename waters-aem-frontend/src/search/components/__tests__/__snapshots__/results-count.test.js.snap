// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature: ResultsCount React Component Scenario: Rendering When keyword is not provided Then the snapshot should match 1`] = `
<div
  className="cmp-search__resultsCount"
  data-locator="results-count"
>
  <div
    class="query-box"
  >
    <span
      class="results"
      dangerouslySetInnerHTML={
        Object {
          "__html": "Showing <span class='count'>24</span> results",
        }
      }
    />
  </div>
  <hr
    className="small-accent-rule"
  />
</div>
`;

exports[`Feature: ResultsCount React Component Scenario: Rendering When keyword is provided with spelling suggestion Then the snapshot should match 1`] = `
<div
  className="cmp-search__resultsCount"
  data-locator="results-count"
>
  <span
    class="results"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Showing <span class='count'>24</span> results for ",
      }
    }
  />
  <span
    className="text-strike"
  >
    milk
  </span>
  <div
    class="query-box"
  >
    <h1
      className="query"
    >
      got milk?
    </h1>
     
     
     
    <span
      class="category"
    >
      in Shop
    </span>
  </div>
  <hr
    className="small-accent-rule"
  />
</div>
`;

exports[`Feature: ResultsCount React Component Scenario: Rendering When keyword is provided without spelling suggestion Then the snapshot should match 1`] = `
<div
  className="cmp-search__resultsCount"
  data-locator="results-count"
>
  <span
    class="results"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Showing <span class='count'>24</span> results for ",
      }
    }
  />
  
  <div
    class="query-box"
  >
    <h1
      className="query"
    >
      milk
    </h1>
     
     
     
    <span
      class="category"
    >
      in Shop
    </span>
  </div>
  <hr
    className="small-accent-rule"
  />
</div>
`;

exports[`Feature: ResultsCount React Component Scenario: Rendering When more than one related keyword is provided Then the snapshot should match 1`] = `
<div
  className="cmp-search__resultsCount"
  data-locator="results-count"
>
  <span
    class="results"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Showing <span class='count'>24</span> results for ",
      }
    }
  />
  <span
    className="text-strike"
  >
    milk
  </span>
  <div
    class="query-box"
  >
    <h1
      className="query"
    >
      got milk?
    </h1>
     
     
     
    <span
      class="category"
    >
      in Shop
    </span>
  </div>
  <hr
    className="small-accent-rule"
  />
  <div
    className="cmp-search__related-suggestions"
  >
    <span
      class="related-searches-text"
    >
      Related Searches
    </span>
    <a
      aria-label="... it does a body good!"
      className="item"
      href="javascript:void(0);"
      onClick={[Function]}
    >
      <div />
      <span>
        ... it does a body good!
      </span>
    </a>
    <a
      aria-label="milkyway"
      className="item"
      href="javascript:void(0);"
      onClick={[Function]}
    >
      <div />
      <span>
        milkyway
      </span>
    </a>
  </div>
</div>
`;

exports[`Feature: ResultsCount React Component Scenario: Rendering When one related keyword is provided Then the snapshot should match 1`] = `
<div
  className="cmp-search__resultsCount"
  data-locator="results-count"
>
  <span
    class="results"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Showing <span class='count'>24</span> results for ",
      }
    }
  />
  <span
    className="text-strike"
  >
    milk
  </span>
  <div
    class="query-box"
  >
    <h1
      className="query"
    >
      got milk?
    </h1>
     
     
     
    <span
      class="category"
    >
      in Shop
    </span>
  </div>
  <hr
    className="small-accent-rule"
  />
  <div
    className="cmp-search__related-suggestions"
  >
    <span
      class="related-searches-text"
    >
      Related Searches
    </span>
    <a
      aria-label="... it does a body good!"
      className="item"
      href="javascript:void(0);"
      onClick={[Function]}
    >
      <div />
      <span>
        ... it does a body good!
      </span>
    </a>
  </div>
</div>
`;

exports[`Feature: ResultsCount React Component Scenario: Rendering When the number of results is higher than the allowed page count Then the snapshot should match 1`] = `
<div
  className="cmp-search__resultsCount"
  data-locator="results-count"
/>
`;

exports[`Feature: ResultsCount React Component Scenario: Rendering When the number of results is less than the allowed page count Then the snapshot should match 1`] = `
<div
  className="cmp-search__resultsCount"
  data-locator="results-count"
/>
`;
